src/app/grid/grid-remote-filtering-sample/remote-filtering-sample.component.ts

```diff
@@ -76,7 +76,10 @@ export class RemoteFilteringSampleComponent implements OnInit {
         }
 
         const virtualizationState = this.grid.virtualizationState;
-        const filteringExpr = this.grid.filteringExpressionsTree.filteringOperands;
+        const filteringExpr = [];
+        this.grid.filteringExpressionsTree.filteringOperands.forEach(operand => {
+            filteringExpr.push(this._remoteService.getFilteringExpressionTreeForColumn(operand))
+        });
```

Comment:
The approach is fine, but I think it would be better to make this transformation in the remote service's `_buildAdvancedFilterExpression` method.